web/src/hooks/useFeatureLocations.ts

```diff
@@ -1,23 +1,33 @@
 import { useCallback } from 'react'
 
-import { createPOIsEndpoint, PoiFeature, PoiModel, prepareFeatureLocations, ReturnType, useLoadAsync } from 'api-client'
+import {
+  createPOIsEndpoint,
+  LocationType,
+  PoiFeature,
+  PoiModel,
+  prepareFeatureLocations,
+  ReturnType,
+  useLoadAsync,
+} from 'api-client'
 
 import { cmsApiBaseUrl } from '../constants/urls'
-import getUserLocation from '../utils/getUserLocation'
 
 type FeatureLocationsType = { features: PoiFeature[]; pois: PoiModel[] }
 
-const useFeatureLocations = (cityCode: string, languageCode: string): ReturnType<FeatureLocationsType> => {
+const useFeatureLocations = (
+  cityCode: string,
+  languageCode: string,
+  coordinates?: LocationType
+): ReturnType<FeatureLocationsType> => {
   const requestFeatureLocations = useCallback(async () => {
-    const { coordinates } = await getUserLocation()
```

Comment:
Why did you remove this from here?

Reply:
To dynamically calculate the distance for each feature. (`currentPoi.getFeature(coordinates)`)